libcontainer/cgroups/fscommon/open.go

```diff
@@ -24,6 +64,29 @@ func OpenFile(dir, file string, flags int) (*os.File, error) {
 		flags |= os.O_TRUNC | os.O_CREATE
 		mode = 0o600
 	}
+	reldir := strings.TrimPrefix(dir, cgroupfsPrefix)
```

Comment:
Seems to require dir = filepath.Clean(filepath.Abs(dir))

Reply:
I would not do it here. This function is called from cgroup managers (with one exception of WriteCgroupFile from runc/libcontainer/utils), and I think we can validate the path in there.

But that is a valid concern, thanks, I'll audit the callers.

Reply:
In fact what happens here is we fall back to old implementation of using securejoin in case `dir` does not start with `/sys/fs/cgroup/`. In case it starts with that, and open2 is available, we use open2 which is guaranteed to not open anything out of /sys/fs/cgroup, so I think we're good here.

Reply:
I think it would be nice if we made this a /little/ less cryptic -- maybe we should make the common interface actually pass the subsystem, cgroup path, and target path -- but aside from that (which we discussed in the other PR and decided against) I don't think this is a problem.

Reply:
@AkihiroSuda so I looked at it one more time -- with the exception of runc/libcontainer/utils.WriteCgroupFile, this function is only used from cgroup managers, and all of them already make sure the path is absolute and resolved (and, in case of cgroupv2, always starts with /sys/fs/cgroup/ which is what we check for here).

Even if it's not, the worst case scenario is we fall back to using securejoin.

So, I don't see it's an issue.

Reply:
Yeah, agreed.